### ISSUE TITLE
Shops overall - request

### ISSUE THREAD

**Issue by PRavenhold** [ForgeEssentials/ForgeEssentials#2028]
a lot of questions regarding shops has been asked on my servers. 

A sell chest - A simple chest where the "owner" puts items into, that sells the stuff and add to the "owners" wallet. 

This has been asked for due to lazyness, /sell is used alot and i'm guessing they get tired of accepting every time. 
-------------------------------------------------------------------------------------
A virtual chest shop / command - (/VFESHop) List each FEShop that exist ingame(use the database) should show amount available and price on each item, would be nice if it also showed amount of currency that the player has. simply left-click once to buy an item, shift+left-click to buy a stack or whats left and drop items in it to sell. A setting on this /VFEShop ShowAvailableOnly true/false, incase you have 200 shops but only 5 of them actually has availble items that players already sold.

**Comment by luacs1998** [comment]
I would probably require inventory events for the sell chest.

What I can do is to add a permission to turn off confirming /sell items to the server...

**Comment by luacs1998** [comment]
Added permission to turn off confirms in https://github.com/ForgeEssentials/ForgeEssentials/commit/af7057744bf009c23de0c333daecad10813de7ca

**Comment by PRavenhold** [comment]
nice :P 

i would so want to help coding the shop :P

**Comment by damonmensch** [comment]
IMO with the intricacy of player shops, especially one's with virtual inventories, this might be better as a separate mod that integrates with FE.

Will I am one to advocate having as few individual mods as possible, it can also potentially cause problems if a single mod gets too complex. Especially if someone wants just the rest of FE, and not the shops, or vice-versa. And yes I am aware that FE is modular in nature, but it's still one large mod where you can't have any part installed without the others installed as well, only disabled.

**Comment by PRavenhold** [comment]
yes but since FE does have the possibility to enable/disable each function then virtualFEShop = disable wouldnt be that hard to change incase you dont want that.. for now im using all functions in FE

**Comment by PRavenhold** [comment]
and im more refering to the FEShops use_stock function instead of using unlimited supply

**Comment by TerdyTheTerd** [comment]
Why not add it to the already existing shop module? If ForgeEssentials already supports admin shops, it only makes sense to add player shops rather than create a whole new mod just for that.

**Comment by luacs1998** [comment]
Thing about it is that we'll need to come up with a good way to tag an "owner" to a shop.

**Comment by damonmensch** [comment]
Best bet is a database and use uuid for the owner field

**Comment by TerdyTheTerd** [comment]
I've seen plugins before that you have to create the sign and place the chest, then while looking at the sign use a command, that way it doesn't check permissions constantly if there's signs and chest being placed near each other until the actual command is used, it also makes getting the owner easier since the owner would be the one using the command, just my two cents.

**Comment by PRavenhold** [comment]
the virtual shop idea wasen't supposed to use player owned shops or a chest. 
was more thinking that it should only show already sold items. like after a player has sold something to an ordinary FEShop, i was more thinking of a gui based  shop, it shouldnt be a problem to drop an item there and refer to the /sell command function. 
same with the buying. so just list the items available(use_stock, im guessing this is already stored in a database) at the moment and refer to /sell and /buy.. 
the only problem would be if the item has no price set yet. maybe a popup or something asking the player if they want to sell it for 0. Could also use the same popup for

**Comment by PRavenhold** [comment]
about the sell chest idea/tag a sign to players own chest with  [Owner] /n <playername>

**Comment by olee** [comment]
About the selling confirmation: You can use the keybinding for questioner to directly confirm questions.
Also you can add a pattern command which could do "/sell" and after that "/yes" to confirm it.
Then you could set this command to a sign / command-block / etc. so a simple click somewhere would be enough.
